### PR TITLE
Process Instance modification gRPC API

### LINKED ISSUE #9659: Create a new gRPC API for the modification command
## Description

We need a new gRPC API to send a modification command from a client to a broker. This includes the extending of the gRPC Protobuf definition and the handling of the requests and responses. As a result, it should be possible to receive a modification command in the gateway, write a modification record on the broker stream, and send the response from the broker back to the client. 
 
The new  gRPC API could look like the following:

```
message ModificationProcessInstanceRequest { 

  message ModificationActivateInstruction {
    string elementId = 1;
    int64 ancestorElementInstanceKey = 2;
    repeated ModificationVariableInstruction variableInstructions = 3;
  }

  message ModificationVariableInstruction {
    string variables = 1;
    string scopeId = 2;
  }

  message ModificationTerminateInstruction {
    int64 elementInstanceKey = 1;
  }

  int64 processInstanceKey = 1;  
  repeated ModificationActivateInstruction activateInstructions = 2;
  repeated ModificationTerminateInstruction terminateInstructions = 3;
}

message ModificationProcessInstanceResponse {
}
``` 

* extend the [gRPC Protobuf definition](https://github.com/camunda/zeebe/blob/main/gateway-protocol/src/main/proto/gateway.proto) for the modification command
* update the [RequestMapper](https://github.com/camunda/zeebe/blob/main/gateway/src/main/java/io/camunda/zeebe/gateway/RequestMapper.java) by the modify request
  * use the process instance key as the key of the command 
* update the [ResponseMapper](https://github.com/camunda/zeebe/blob/main/gateway/src/main/java/io/camunda/zeebe/gateway/ResponseMapper.java) by the modify response

Blocked by #9638

### PR DESCRIPTION
**Pull request by remcowesterhoud**
## Description

This PR adds the PI modification RPC toe the gateway. It maps the request and response.

Note: I have named the RPC slightly different from what was mentioned in the issue. I believe `ModifyProcessInstanceRequest` is more in line with the other names than `ModificationProcessInstanceRequest`.

## Related issues

closes #9659

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/request/BrokerModifyProcessInstanceRequest.java
Comment (korthout) [inline gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/request/BrokerModifyProcessInstanceRequest.java:24]: 🔧 class can be `final`

File: gateway-protocol/src/main/proto/gateway.proto
Comment (korthout) [inline gateway-protocol/src/main/proto/gateway.proto:420]: 🔧 Officially this field is optional, but we generally don't define that in the API. However, it might be useful to describe when it's needed or what is achieved by setting it to `-1`. For example, we did this with `version` of `CreateProcessInstanceRequest`.

File: gateway-protocol/src/main/proto/gateway.proto
Comment (korthout) [inline gateway-protocol/src/main/proto/gateway.proto:426]: 🔧 Please copy the definition of variables from `CreateProcessInstanceRequest`

File: gateway-protocol/src/main/proto/gateway.proto
Comment (korthout) [inline gateway-protocol/src/main/proto/gateway.proto:429]: 🔧 This field is also optional, so users can set it to `-1`. Let's describe what happens in that case. 

>(optional) the element id of the scope where the variables should be created in, otherwise the variables are global for the process instance

See https://github.com/camunda/zeebe/issues/9638

**Comment by remcowesterhoud** [comment]
FYI, CodeQL will fail because it is scanning generated code. Unfortunately there is no easy way to exclude these files 😞 

#9732

**Review by korthout — APPROVED** [review]
👍 Thanks for this @remcowesterhoud

❌ Please don't forget to run `mvn clean install` and commit the changes to `proto.lock` and the license headers.

I've added a few suggestions, please have a look. None of these have to block merging the PR after you've added the changes to the `proto.lock` file.

So LGTM 👍

File: gateway-protocol/src/main/proto/gateway.proto
Comment (remcowesterhoud) [reply gateway-protocol/src/main/proto/gateway.proto:410]: I don't feel like this extra part makes it less clear. If anything it describes better what the instruction is used for.

File: gateway-protocol/src/main/proto/gateway.proto
Comment (remcowesterhoud) [reply gateway-protocol/src/main/proto/gateway.proto:420]: What's the reason we don't define that fields are optional in the API?

Comment (remcowesterhoud) [reply gateway-protocol/src/main/proto/gateway.proto:420]: Isn't setting something to -1 different than leaving the field out entirely?

File: gateway-protocol/src/main/proto/gateway.proto
Comment (remcowesterhoud) [reply gateway-protocol/src/main/proto/gateway.proto:632]: Thanks, I forgot to add those 😄

**Comment by remcowesterhoud** [comment]
@korthout Please have a look at https://github.com/camunda/zeebe/pull/9873#discussion_r929773929 before I merge it.

File: gateway-protocol/src/main/proto/gateway.proto
Comment (korthout) [reply gateway-protocol/src/main/proto/gateway.proto]: 🤷 I have no clue why we don't actually, personally I don't mind it much

**Review by korthout — APPROVED** [review]
Nice work 💯 LGTM

**Comment by remcowesterhoud** [comment]
bors merge

**Pull request merged**
